WooPay naming/styling updates

malithsen opened the pull request:
Fixes 635-gh-Automattic/woopay

#### Changes proposed in this Pull Request

* Update registration update copy (see 635-gh-Automattic/woopay)
* Replace Platform Checkout with WooPay in the UI
* Replace the placeholder Woo logo on the settings page to use the WooPay logo
* Use production URLs as the default in `PLATFORM_CHECKOUT_FRONTEND_HOST` and `PLATFORM_CHECKOUT_HOST`

Context: p1649353944790879-slack-C022WMN88KG, pdpOw2-yo-p2
<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Make sure platform checkout is enabled for your merchant site.
* Ensure that WooPay registration UI matches the updated design in pdpOw2-yo-p2. Note that the link in "powered by WooPay" is not set on this PR, this will be set later.
* When `PLATFORM_CHECKOUT_FRONTEND_HOST` and `PLATFORM_CHECKOUT_HOST` is not set, WCPay attempts to load the OTP iframe from `woo.app`
* WooCommerce -> Settings -> Payments -> Express Checkout page uses the correct logo cO37XZJMyjQweZ6RhEeDhX-fi-31%3A25535

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _Add link here / 'QA Testing Not Applicable'_

bborman22 commented on client/components/platform-checkout/save-user/agreement.js:
```diff
@@ -10,24 +10,24 @@ const Agreement = () => {
-							href="https://automattic.com/privacy/"
 							target="_blank"
-							rel="noreferrer"
+							href="https://href.li/?https://automattic.com/privacy/"
```
I think this might just be a copy/paste artifact to address?

bborman22 reviewed (changes requested):
This all looked good! I found one small typo and one link issue, but otherwise everything matched the designs and descriptions I saw in discussion and designs.

I did see that we reference platform checkout in the [payment settings API parameter descriptions](https://github.com/Automattic/woocommerce-payments/blob/56595f45b9e083c2d3e2374c7e528199267b7980/includes/admin/class-wc-rest-payments-settings-controller.php#L202). I for sure don't feel that's a high priority to address (and maybe shouldn't even be changed?) and should not block anything, but figured I'd mention it in case this would be a good time to update that?

After the link and typo are addressed this LGTM!

bborman22 reviewed (approved):
Thanks for those fixes, LGTM!

malithsen commented:
Thanks for the review @bborman22, fixed the link and typo in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/f8532c5d359aeacb1fad6c41a457db88537ebbf1

> I did see that we reference platform checkout in the [payment settings API parameter descriptions](https://github.com/Automattic/woocommerce-payments/blob/56595f45b9e083c2d3e2374c7e528199267b7980/includes/admin/class-wc-rest-payments-settings-controller.php#L202). I for sure don't feel that's a high priority to address (and maybe shouldn't even be changed?)

I missed this initially, I adjusted the description in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/a42d5a1928062f534fe197ea0862fc6656c3a8da but I am not sure if it ever shows up in the UI or where it shows up.

leonardola commented on client/components/platform-checkout/save-user/agreement.js:
```diff
@@ -10,24 +10,24 @@ const Agreement = () => {
 			{ interpolateComponents( {
 				mixedString: __(
-					'By entering your phone number and completing your purchase, you will create a Platform Checkout account and agree to {{termsOfService/}} and {{privacyPolicy/}}.',
+					'By entering your phone number and completing your purchase, you will create a WooPay account and agree to {{termsOfService/}} and {{privacyPolicy/}}.',
```
this should read:

By entering your mobile phone number and completing your purchase, you will create a WooPay account, and you agree to the Terms of Service and Privacy Policy

malithsen replied:
Updated in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/b3da826f56743af9f316ba7a8d5c8b05f2e95664

leonardola reviewed (changes requested):
@malithsen I noticed that the checkbox uses the default browser styling while the Updated design uses a WooCommerce color.
How it actually looks:
<img width="336" alt="Screen Shot 2022-04-08 at 2 16 47 PM" src="https://user-images.githubusercontent.com/6342964/162490718-1aa2ed65-4b67-45de-9f28-e4455124f8ed.png">

How it should look:
<img width="275" alt="Screen Shot 2022-04-08 at 2 19 01 PM" src="https://user-images.githubusercontent.com/6342964/162490769-5af72a31-8713-427f-a49e-3d19db804e5b.png">

Also the label "Mobile phone number" is missing before the phone number input.

How it actually looks:
<img width="336" alt="Screen Shot 2022-04-08 at 2 19 46 PM" src="https://user-images.githubusercontent.com/6342964/162491022-9bec3cd8-b2dc-4fbe-a2a6-40bf04944e00.png">


How it should look:
<img width="254" alt="Screen Shot 2022-04-08 at 2 19 51 PM" src="https://user-images.githubusercontent.com/6342964/162490890-3da99e0e-7515-4d97-b546-530e10f39781.png">
 
I also left a comment about a string that is different from the one levin posted

malithsen commented:
> Also the label "Mobile phone number" is missing before the phone number input.

Nice catch! Updated in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/49a5539daef82a64c1f7d8cc9f94002e7598ebaa

> I noticed that the checkbox uses the default browser styling while the Updated design uses a WooCommerce color.

I played this for a bit but couldn't get it exactly right. I'd like to separate this out to a different issue to unblock the merge of the rest of the changes here.

leonardola reviewed (approved):
This looks perfect now. Thanks for working on this one. :shipit:
